### PR TITLE
fix: MC flags size and serialization #3134

### PR DESCRIPTION
**Pull request by BorysTheDev**
fix for #3134. MC_flags is uint32_t now and I added serialization and deserialization for it.

### CODE REVIEW

**Comment by kostasrim** [comment]
plz edit the description + reference the issue it fixes. I will review it once the tests are green :)

File: src/server/string_family.cc [hunk 1]
```diff
@@ -782,7 +782,7 @@ void StringFamily::Set(CmdArgList args, ConnectionContext* cntx) {
 
       tie(sparams.expire_after_ms, ignore) = expiry.Calculate(GetCurrentTimeMs(), true);
     } else if (parser.Check("_MCFLAGS").ExpectTail(1)) {
-      sparams.memcache_flags = parser.Next<uint16_t>();
+      sparams.memcache_flags = parser.Next<uint32_t>();
```
Comment (romange) [inline src/server/string_family.cc:785]: Good catch

File: tests/dragonfly/replication_test.py [hunk 2]
```diff
@@ -2511,3 +2511,35 @@ async def check_if_empty():
     assert await c_replica.execute_command(f"dbsize") == 0
 
     await close_clients(c_master, c_replica)
+
+
+async def test_replicating_mc_flags(df_factory):
```
Comment (romange) [inline tests/dragonfly/replication_test.py:2516]: looks good but why not test it via rdb_test.cc ?

**Review by romange — APPROVED** [review]

File: tests/dragonfly/replication_test.py (hunk 2 above)
Comment (romange) [reply tests/dragonfly/replication_test.py:2516]: no, it's fine. next time use rdb_test  - it's easier to debug via unit tests :)

**Comment by romange** [comment]
LGTM!

**Pull request merged**